data.md

```diff
@@ -101,17 +101,18 @@ These can be used for pattern-matching on the LHS of rules as well (`macro` attr
                   | #rangeSFixed  ( Int , Int , Int )
                   | #rangeUFixed  ( Int , Int , Int )
                   | #rangeAddress ( Int )
-                  | #rangeBytes   ( Int , Int )
- // -------------------------------------------
-    rule #rangeSInt    ( 128 ,      X ) => #range ( minSInt128      <= X <= maxSInt128      ) [macro]
-    rule #rangeSInt    ( 256 ,      X ) => #range ( minSInt256      <= X <= maxSInt256      ) [macro]
-    rule #rangeUInt    (  48 ,      X ) => #range ( minUInt48       <= X <= maxUInt48       ) [macro]
-    rule #rangeUInt    ( 128 ,      X ) => #range ( minUInt128      <= X <= maxUInt128      ) [macro]
-    rule #rangeUInt    ( 256 ,      X ) => #range ( minUInt256      <= X <= maxUInt256      ) [macro]
-    rule #rangeSFixed  ( 128 , 10 , X ) => #range ( minSFixed128x10 <= X <= maxSFixed128x10 ) [macro]
-    rule #rangeUFixed  ( 128 , 10 , X ) => #range ( minUFixed128x10 <= X <= maxUFixed128x10 ) [macro]
-    rule #rangeAddress (            X ) => #range ( minUInt160      <= X <= maxUInt160      ) [macro]
-    rule #rangeBytes   (  32 ,      X ) => #range ( minUInt256      <= X <= maxUInt256      ) [macro]
+                  | #rangeBytes   ( Int , Int ) [function]
+ // ------------------------------------------------------
+    rule #rangeSInt    ( 128 ,      X ) => #range ( minSInt128      <= X <= maxSInt128        ) [macro]
+    rule #rangeSInt    ( 256 ,      X ) => #range ( minSInt256      <= X <= maxSInt256        ) [macro]
+    rule #rangeUInt    (   8 ,      X ) => #range ( minUInt8        <= X <= maxUInt8          ) [macro]
+    rule #rangeUInt    (  48 ,      X ) => #range ( minUInt48       <= X <= maxUInt48         ) [macro]
+    rule #rangeUInt    ( 128 ,      X ) => #range ( minUInt128      <= X <= maxUInt128        ) [macro]
+    rule #rangeUInt    ( 256 ,      X ) => #range ( minUInt256      <= X <= maxUInt256        ) [macro]
+    rule #rangeSFixed  ( 128 , 10 , X ) => #range ( minSFixed128x10 <= X <= maxSFixed128x10   ) [macro]
+    rule #rangeUFixed  ( 128 , 10 , X ) => #range ( minUFixed128x10 <= X <= maxUFixed128x10   ) [macro]
+    rule #rangeAddress (            X ) => #range ( minUInt160      <= X <= maxUInt160        ) [macro]
+    rule #rangeBytes   (   N ,      X ) => #range ( 0               <= X <= 256 ^Int N -Int 1 ) // 2^(N * 8) - 1 = 256^N - 1
```

Comment:
@denis-bogdanas (and @daejunpark) why is this not a `macro` like the other rules?

Reply:
This can be macro as well. Regardless of whether N is constant or variable, the rule can safely apply. I'll fix this. Thanks for pointing this out.